### PR TITLE
Create and connect ban service

### PR DESCRIPTION
**Pull request by AlB1111122**
## Summary
https://trello.com/c/1sHzxW4d
https://trello.com/c/sSMWA65p
change network of userservice to an external network so http reqs can be sent between
create a ban service
create endpoints for:
 - adding a new ban for shops and users
 - getting all bans for a specific shop or user
 - checking which resources needed to be unhidden on their own services and doing it
 - getting all of the bans
 - edit env.sh to env.py to improve window/linux compatibiltiy
store env vars on spring cloud config
`python3 env.py`
connected the banning endpoints to their respective srevies
## How to verify

`../gradlew build`
`docker build -t banservice .`
`docker compose up`
test out the endpoints

### CODE REVIEW

**Comment by AlB1111122** [comment]
[Create Ban service](https://trello.com/c/1sHzxW4d/18-create-ban-service)

**Review by tadhgryan2012 — DISMISSED** [review]
Looks good!

**Review by tadhgryan2012 — DISMISSED** [review]
lgtm

**Review by oOoOoOo0o — DISMISSED** [review]
Very good work.
I Tested every endpoint and almost everything works, a few minor changes needed.
And of course when environment variables are merged in that will be configured so as to prevent using the same port twice in the case of the db.

File: BanService/build.gradle [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    mavenCentral()
+}
+
+dependencies {
+    implementation 'org.springframework.boot:spring-boot-starter-web'
+    testImplementation 'org.springframework.boot:spring-boot-starter-test'
+    testRuntimeOnly 'org.junit.platform:junit-platform-launcher'
+    implementation 'org.springframework.boot:spring-boot-starter-jdbc'
+    implementation 'mysql:mysql-connector-java:8.0.33'
+}
```
Comment (oOoOoOo0o) [inline BanService/build.gradle:27]: Need to add cloud starter config
```
implementation 'org.springframework.cloud:spring-cloud-starter-config'
```

File: BanService/build.gradle [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
```
Comment (oOoOoOo0o) [inline BanService/build.gradle:28]: And add
```
dependencyManagement {
    imports {
        mavenBom "org.springframework.cloud:spring-cloud-dependencies:2023.0.3"
    }
}
```

File: BanService/src/main/java/CS4337/Project/BanService.java [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+
+@SpringBootApplication
+@RestController
+public class BanService {
+  // TODO: replace with env vars
+  private String userServiceUrl = "http://user-service:9090/user/ban/{id}";
+  // fix this later
+  private String shopServiceUrl = "http://shop-service:8080/shopItem/ban/{id}";
```
Comment (oOoOoOo0o) [inline BanService/src/main/java/CS4337/Project/BanService.java:32]: This endpoint doesn't exist in shop service, maybe we can add it in this PR, here is an example:
```java
  @PutMapping("shopItem/ban/{id}")
  public ResponseEntity<Map<String, Object>> banShopItem(
      @PathVariable("id") int id, @RequestBody Map<String, String> requestBody) {
    try {
      String hiddenStr = requestBody.get("isHidden");
      Boolean hidden = (hiddenStr == "true" ? true : false);
      String banStatement = "UPDATE `ShopItem` SET isHidden = ? WHERE id = ?";
      jdbcTemplate.update(banStatement, hidden, id);
      return ResponseEntity.ok(Map.of("Shop item banned", hidden));
    } catch (DataAccessException e) {
      return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(Map.of("error", e.getMessage()));
    }
  }
  ```

**Review by oOoOoOo0o — CHANGES_REQUESTED** [review]

**Comment by AlB1111122** [comment]
[set up calls from banservice to other services](https://trello.com/c/sSMWA65p/23-set-up-calls-from-banservice-to-other-services)

**Review by oOoOoOo0o — APPROVED** [review]

**Review by tadhgryan2012 — APPROVED** [review]
LGTM

**Review by tadhgryan2012 — APPROVED** [review]
Please merge

**Pull request merged**